### PR TITLE
[1.13-backport] ipsec: Fix unencrypted traffic when IPsec is used with L7 egress proxy

### PR DESCRIPTION
**Pull request by jschwinger233**
- [x] https://github.com/cilium/cilium/pull/32683 @jschwinger233 
- [x] https://github.com/cilium/cilium/pull/32906 @julianwiedmann 

```upstream-prs
 32683 32906
```

### CODE REVIEW

**Comment by jschwinger233** [comment]
/test-backport-1.13

**Comment by jschwinger233** [comment]
/test-backport-1.13

**Comment by jschwinger233** [comment]
/test-1.17-4.19

**Comment by jschwinger233** [comment]
/test-backport-1.13

**Comment by julianwiedmann** [comment]
:wave: when dropping the leak detection patches, could you please also pick up https://github.com/cilium/cilium/pull/32906 in this PR?

**Comment by jschwinger233** [comment]
@julianwiedmann No problem.

**Comment by jschwinger233** [comment]
ci-ipsec-e2e with leak detection looks fine except for the known racing issue: https://github.com/cilium/cilium/actions/runs/9414645480/job/25933978400

**Comment by jschwinger233** [comment]
/test-backport-1.13

File: .github/workflows/tests-ipsec-upgrade.yaml [hunk 1]
```diff
@@ -393,6 +393,13 @@ jobs:
             kubectl get pods --all-namespaces -o wide
             kubectl -n kube-system exec daemonset/cilium -c cilium-agent -- cilium status
 
+            # 1.12 is EOL, we have to run commands manually to clean up stale routing rules
+            cilium_pods=$(kubectl -nkube-system get po -l k8s-app=cilium --no-headers -o custom-columns=":metadata.name")
+            for cilium_pod in $cilium_pods; do
+              kubectl -nkube-system exec $cilium_pod -- ip -4 rule delete fwmark 0xB00/0xF00 lookup 2005 2>/dev/null || true
+              kubectl -nkube-system exec $cilium_pod -- ip -6 rule delete fwmark 0xB00/0xF00 lookup 2005 2>/dev/null || true
+            done
+
```
Comment (julianwiedmann) [inline .github/workflows/tests-ipsec-upgrade.yaml:402]: This is something we only need for `[ "${{ matrix.mode }}" = "minor" ]`, right? For the `patch` test we already have everything what's needed in `v1.13`.

(if you agree, I think it's fine to fix that up in a follow-up PR. No need to re-run the whole CI for it)

Comment (jschwinger233) [reply .github/workflows/tests-ipsec-upgrade.yaml:402]: (I didn't receive email notification for this comment :fearful: )

Agree, thank you :+1:

**Comment by jschwinger233** [comment]
/test-1.19-4.19

**Comment by jschwinger233** [comment]
/test-1.24-4.19

**Comment by jschwinger233** [comment]
/test-1.25-4.19

**Review by brlbil — APPROVED** [review]

**Review by julianwiedmann — APPROVED** [review]
thank you!

**Review by dylandreimerink — APPROVED** [review]

**Pull request merged**